python/message.c

```diff
@@ -1031,34 +1033,50 @@ static PyObject* PyUpb_CMessage_Clear(PyUpb_CMessage* self, PyObject* args) {
   Py_RETURN_NONE;
 }
 
-static void PyUpb_CMessage_AbandonField(PyUpb_CMessage* self,
-                                        const upb_fielddef* f) {
-  if (self->unset_subobj_map && upb_fielddef_issubmsg(f) &&
-      !upb_fielddef_isseq(f)) {
-    PyObject* sub = PyUpb_WeakMap_Get(self->unset_subobj_map, f);
-    if (sub) {
-      PyUpb_CMessage_AssureReified((PyUpb_CMessage*)sub);
+void PyUpb_CMessage_DoClearField(PyObject* _self, const upb_fielddef* f) {
+  PyUpb_CMessage* self = (void*)_self;
+  PyUpb_CMessage_AssureReified((PyUpb_CMessage*)self);
```

Comment:
This should be `EnsureRefified` not `AssureReified` since it will actually do the work to change it if needed

Reply:
Done.